src/lib/methods/batch/batch-write.ts

```diff
@@ -0,0 +1,86 @@
+import { Executable } from '../executable';
+import { DynamoDB } from '../../dynamodb';
+import { BatchItem } from './batch-item';
+import { BatchMethod } from './batch-method';
+import { BatchWriteItemInput } from 'aws-sdk/clients/dynamodb';
+import pRetry from 'p-retry';
+
+export class BatchWrite extends BatchMethod implements Executable {
+
+	constructor(
+		dynamodb: DynamoDB,
+		private items: BatchItem[],
+		private retryCount = 5,
+		private randomizeRetry = true) {
+		super(dynamodb);
+	}
+	/**
+	 * Initialize the `InsertItem` object.
+	 */
+	initialize(): BatchWrite {
```

Comment:
Just wondering. Why do we have the `initialize` here? Can't we move the code from the `initialize` to the `buildRawQuery`?

Also, is there any reason the request is assigned to `this.params`? It feels like `this.params` might not even have to exist?

Reply:
Yes I agree we should remove initialize and replace it with buildRawQuery. I think I was confusing the 2 when looking at the other classes. The this.params approach was used because I wanted it to be similar to the other classes. I replaced it now with the suggested changes and made buildRawQuery return BatchWriteItemInput which keeps the typechecking.